include/engine/datafacade/internal_datafacade.hpp

```diff
@@ -215,31 +215,36 @@ class InternalDataFacade final : public BaseDataFacade
         }
     }
 
-    void LoadDatasourceInfo(const boost::filesystem::path &datasource_names_file,
-                            const boost::filesystem::path &datasource_indexes_file)
+    void LoadDatasourceInfo(const std::string &datasource_names_file,
+                            const std::string &datasource_indexes_file)
```

Comment:
Why suddenly strings? boost::fs::path should be fine as we're using boost::fs::ifstream anyway which is the same as std::ifstream but in addition has a ctor from boost::fs::path.